lib/ansible/galaxy/collection.py

```diff
@@ -1380,15 +1402,21 @@ def _extract_tar_file(tar, filename, b_dest, b_temp_path, expected_hash=None):
             # makes sure we create the parent directory even if it wasn't set in the metadata.
             os.makedirs(b_parent_dir, mode=0o0755)
 
-        shutil.move(to_bytes(tmpfile_obj.name, errors='surrogate_or_strict'), b_dest_filepath)
+        if tar_member.type == tarfile.SYMTYPE:
+            # TODO: Should we validate the link name is inside the collection. We already check the hash matches the
+            # manifest which typically fails because the file wasn't known at build time.
```

Comment:
It's probably a good idea, to avoid creating links to `/etc/passwd` or other "funny" files. (Not that I know how to abuse that, but better to prevent this before accidentally allowing it and finding out later that it leads to big problems.)

Reply:
So the reason why I didn't implement it, and added the TODO to discuss, is because this part of the code will only be reached if the checksum in the `FILES.json` match the checksum of the link target. So for a symlink to be created that points to `/etc/passwd`, the collection artifact needs to already have the correct hash.

While that doesn't stop someone from editing the artifact after it has been created, they would already have to have access to that file and therefore don't need the symlink to expose it.